### PR TITLE
Update unit test for PBO

### PR DESCRIPTION
**Pull request by teytaud**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the [**CONTRIBUTING**](https://facebookresearch.github.io/nevergrad/contributing.html) document and completed the CLA (see [**CLA**](https://facebookresearch.github.io/nevergrad/contributing.html#contributor-license-agreement-cla)).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

File: nevergrad/functions/iohprofiler/core.py [hunk 1]
```diff
@@ -138,4 +138,6 @@ def __init__(
 
     def _evaluation_internal(self, x: np.ndarray) -> float:
         assert len(x) == self.f_internal.number_of_variables
+        for i in range(len(x)):
+            assert x[i] in [0, 1], f"Pb with input {x} in PBO: not binary."
```
Comment (jrapin) [inline nevergrad/functions/iohprofiler/core.py:142]:
```suggestion
        assert not set(x) - {0, 1},  f"Pb with input {x} in PBO: not binary."
```

File: nevergrad/functions/iohprofiler/test_core.py [hunk 2]
```diff
@@ -7,18 +7,31 @@
...
 def test_PBO(fid: int) -> None:
-    try:
-        func = core.PBOFunction(fid, 0, 16)
-    except ModuleNotFoundError:
-        raise SkipTest("IOH is not installed")
-    x = func.parametrization.sample()
-    value = func(x.value)
-    assert isinstance(value, float), "All output of the iohprofiler-functions should be float"
-    assert np.isfinite(value)
+    for iid in range(1, 5):
```
Comment (jrapin) [inline nevergrad/functions/iohprofiler/test_core.py:15]: use pytest.mark.parametrize for iid as well

**Review by jrapin — CHANGES_REQUESTED** [review]

File: nevergrad/functions/iohprofiler/test_core.py (hunk 2 above)
Comment (teytaud) [reply nevergrad/functions/iohprofiler/test_core.py]: done

**Review by jrapin — APPROVED** [review]

File: nevergrad/functions/iohprofiler/test_core.py [hunk 3]
```diff
@@ -7,18 +7,31 @@
...
+        func = core.PBOFunction(fid, iid, 16)
     except ModuleNotFoundError:
         raise SkipTest("IOH is not installed")
-    x = func.parametrization.sample()
-    value = func(x.value)
-    assert isinstance(value, float), "All output of the iohprofiler-functions should be float"
-    assert np.isfinite(value)
+    for _ in range(30):
+        x = func.parametrization.sample()
+        for i in range(len(x.value)):
```
Comment (jrapin) [inline nevergrad/functions/iohprofiler/test_core.py:23]: `assert not set(x.value) - {0, 1}, f"Non binary sample {x}.`
should be enough

**Pull request merged**